zerver/worker/queue_processors.py

```diff
@@ -342,6 +345,56 @@ def consume(self, event):
         mirror_email(email.message_from_string(event["message"].encode("utf-8")),
                      rcpt_to=event["rcpt_to"], pre_checked=True)
 
+@assign_queue('outhook_worker')
+class OutgoingWebhookWorker(QueueProcessingWorker):
+
+    def send_response_message(self, bot_email, trigger_message, response_message):
+        recipient_type_name = trigger_message['type']
+        if recipient_type_name == 'stream':
+            recipients = trigger_message['display_recipient']
+            internal_send_message(bot_email, recipient_type_name, recipients,
+                                  trigger_message['subject'], response_message)
+        else:
+            # Private message; only send if the bot is there in the recipients
+            trigger_message_recipients = [recipient['email'] for recipient in trigger_message['display_recipient']]
+            if bot_email in trigger_message_recipients:
+                recipients = ','.join(trigger_message_recipients)
+                internal_send_message(bot_email, recipient_type_name, recipients,
+                              trigger_message['subject'], response_message)
+
+    def retry_trigger(self, bot_handler, event):
+        event['retry'] +=1
+        if event['retry'] > 3:
+            response_message = "Maximum retries exceeded"
+            bot_email = event['bot_email']
+            command = event['command']
+            self.send_response_message(bot_email, event['trigger_message'], response_message)
+            logging.warning("Maximum retries exceeded for trigger:%s event:%s" % (bot_email, command))
+        else:
+            queue_json_publish("outhook_worker", event, lambda x: None)
+
+    def consume(self, event):
+        bot_email = event['bot_email']
+        command = event['command']
+        trigger_message = event['message']
+        realm = get_realm(trigger_message["sender_domain"])
+
+        bot_handler = get_outgoing_webhook_bot_handler(bot_email, realm)
+
+        # We have to catch any requests exception and handle them appropriatly
+        try:
+            bot_response_message = bot_handler.trigger(command)
+            if bot_response_message is not None:
+                self.send_response_message(bot_email, trigger_message, bot_response_message)
+
+        except requests.exceptions.Timeout:
+            self.retry_trigger(bot_handler, event)
+
+        except requests.exceptions.RequestException as e:
+            response_message = "An exception occured! See the logs for more information."
+            self.send_response_message(bot_email, trigger_message, response_message)
+            logging.exception("Outhook trigger failed:\n %s" % (e,))
```

Comment:
Check `api/zulip/__init__.py` for the other exceptions requests often throws (or at least used to) -- we may want to add some more to this set.